### DISCUSSION TITLE
Trying to add a new device

### DISCUSSION THREAD

**Discussion by mplab-casa** [make-all/tuya-local#2397] [Q&A] [answered]
Hello everyone, I am trying to add a new device but I am having some difficulties.
The device is Rowenta X-Plorer S75.
This is a clone of an existing device(Tefal X-plorer Serie 75 Animal), so it should be easy.

I connected to the homeassistant instance, went inside the folder “custom_components/tuya_local/devices” copied the file tefal_xplorer_vacuum.yml and changed the values of name and products.

However, this does not work, what am I doing wrong?

From homeassistant I get this log:
`Device matches garage_door_opener with quality of 6%. DPS: {"updated_at": 1729014309.8481982, "101": false, "102": false, "103": true, "105": "chargring", "106": 99, "107": 0, "108": 0, "109": "max", "113": false, "116": 88178, "117": 1465, "118": 60, "119": 88178, "120": 88178, "121": 88178, "122": 0, "137": true, "144": false}`

These are the device details:
`    {
      "active_time": 1716028762,
      "bind_space_id": "188445559",
      "category": "sd",
      "create_time": 1716028762,
      "custom_name": "",
      "icon": "smart/icon/ay15748409168332lI6v/9866525ed4c15fa8e4abcf2e5f53735f.png",
      "id": "XXXXXXXXXXXXXXXXXXXXX",
      "ip": "XX.XX.XX.XX",
      "is_online": true,
      "lat": "XX",
      "local_key": "eK|r8f8blQyENNok",
      "lon": "XX",
      "model": "L5T",
      "name": "X-Plorer Serie 75",
      "product_id": "tw2o6dntmqfpwsc7",
      "product_name": "🟢 S75 (Tefal/Rowenta)",
      "sub": false,
      "time_zone": "+XX:00",
      "update_time": 1726485486,
      "uuid": "XXXXXXXXXXXXXXXXXX"
    }
`

This Query Things Data Model:
`    "model": "{\"modelId\":\"000001kbcl\",\"services\":[{\"actions\":[],\"code\":\"\",\"description\":\"\",\"events\":[],\"name\":\"默认服务\",\"properties\":[{\"abilityId\":1,\"accessMode\":\"rw\",\"code\":\"power\",\"description\":\"【非必选】\\n\\\"可选择无开关机功能。\\n开关机控制。\\n下发： 发送控制指令给主机，主机依据指令状态进行开关机操作。\\n上报：\\n当主机使用遥控器关机或者长时间静止进入关机状态后，主机会上报一个关机数据。\\n当主机处于关机状态时，通过遥控器或者触摸主机的按键，主机会进入开机状态，主机会上报一个开机数据。\\\"\\n\",\"extensions\":{\"iconName\":\"icon-dp_power2\"},\"name\":\"电源开关\",\"typeSpec\":{\"type\":\"bool\"}},{\"abilityId\":2,\"accessMode\":\"rw\",\"code\":\"power_go\",\"description\":\"【必选】\\n\",\"extensions\":{\"iconName\":\"icon-dp_power\"},\"name\":\"电源\",\"typeSpec\":{\"type\":\"bool\"}},{\"abilityId\":3,\"accessMode\":\"rw\",\"code\":\"mode\",\"description\":\"【必选】枚举值按需选择，不需要的可删除\\nstandby：待机模式，random：随机模式，smart：自动清扫， wall_follow：沿边清扫 ，mop:拖地功能 ， spiral:定点清扫，left_spiral:左螺旋，right_spiral:右螺旋，right_bow：右⼸弓字，left_bow：左⼸弓字，partial_bow：局部弓字, chargego:回充模式\\n\",\"extensions\":{\"iconName\":\"icon-dp_mode\"},\"name\":\"清扫模式\",\"typeSpec\":{\"type\":\"enum\",\"range\":[\"standby\",\"random\",\"smart\",\"wall_follow\",\"mop\",\"spiral\",\"left_spiral\",\"right_spiral\",\"right_bow\",\"left_bow\",\"partial_bow\",\"chargego\"]}},{\"abilityId\":13,\"accessMode\":\"rw\",\"code\":\"seek\",\"description\":\"【非必选】\\n\",\"extensions\":{\"iconName\":\"icon-dp_curve\"},\"name\":\"寻找机器\",\"typeSpec\":{\"type\":\"bool\"}},{\"abilityId\":101,\"accessMode\":\"rw\",\"code\":\"clean_switch\",\"description\":\"1开始全图清扫，0结束清扫\",\"name\":\"清扫\",\"typeSpec\":{\"type\":\"bool\"}},{\"abilityId\":102,\"accessMode\":\"rw\",\"code\":\"pause_switch\",\"description\":\"1暂停，0继续\",\"name\":\"暂停\",\"typeSpec\":{\"type\":\"bool\"}},{\"abilityId\":103,\"accessMode\":\"rw\",\"code\":\"charge_switch\",\"description\":\"\",\"name\":\"回充\",\"typeSpec\":{\"type\":\"bool\"}},{\"abilityId\":104,\"accessMode\":\"rw\",\"code\":\"clean_mode\",\"description\":\"下发表示开始对应清扫，上传表示当前的清扫类型\",\"name\":\"清扫模式\",\"typeSpec\":{\"type\":\"enum\",\"range\":[\"smart\",\"pose\",\"zone\",\"backcharge\",\"curpointing\",\"selectroom\"]}},{\"abilityId\":105,\"accessMode\":\"rw\",\"code\":\"robot_state\",\"description\":\"\",\"name\":\"当前状态：\",\"typeSpec\":{\"type\":\"enum\",\"range\":[\"idle\",\"pointing\",\"areaing\",\"totaling\",\"sweep\",\"mop\",\"fault\",\"pause\",\"chargring\",\"tocharge\",\"fullcharge\",\"remotectl\",\"dormant\",\"curpointing\",\"selectroom\"]}},{\"abilityId\":106,\"accessMode\":\"rw\",\"code\":\"battery\",\"description\":\"\",\"name\":\"剩余电量\",\"typeSpec\":{\"type\":\"value\",\"max\":100,\"min\":0,\"scale\":0,\"step\":1,\"unit\":\"%\"}},{\"abilityId\":107,\"accessMode\":\"rw\",\"code\":\"cur_clean_time\",\"description\":\"\",\"name\":\"清扫时间\",\"typeSpec\":{\"type\":\"value\",\"max\":2147483647,\"min\":0,\"scale\":0,\"step\":1,\"unit\":\"秒\"}},{\"abilityId\":108,\"accessMode\":\"rw\",\"code\":\"cur_clean_area\",\"description\":\"\",\"name\":\"清扫面积\",\"typeSpec\":{\"type\":\"value\",\"max\":2147483647,\"min\":0,\"scale\":0,\"step\":1,\"unit\":\"\"}},{\"abilityId\":109,\"accessMode\":\"rw\",\"code\":\"fan_mode\",\"description\":\"\",\"name\":\"吸力\",\"typeSpec\":{\"type\":\"enum\",\"range\":[\"quiet\",\"auto\",\"strong\",\"max\"]}},{\"abilityId\":111,\"accessMode\":\"rw\",\"code\":\"remote_ctrl\",\"description\":\"\",\"name\":\"远程遥控\",\"typeSpec\":{\"type\":\"enum\",\"range\":[\"forward\",\"backward\",\"left\",\"right\",\"stop\"]}},{\"abilityId\":112,\"accessMode\":\"rw\",\"code\":\"seek_robot\",\"description\":\"\",\"name\":\"定位机器人\",\"typeSpec\":{\"type\":\"bool\"}},{\"abilityId\":113,\"accessMode\":\"rw\",\"code\":\"disturb_switch\",\"description\":\"\",\"name\":\"勿扰模式\",\"typeSpec\":{\"type\":\"bool\"}},{\"abilityId\":115,\"accessMode\":\"wr\",\"code\":\"material_reset\",\"description\":\"\",\"name\":\"耗材与维护\",\"typeSpec\":{\"type\":\"enum\",\"range\":[\"gettime\",\"resetsidebrush\",\"resetmainbrush\",\"resetfilter\"]}},{\"abilityId\":116,\"accessMode\":\"rw\",\"code\":\"total_clean_time\",\"description\":\"\",\"name\":\"总时长\",\"typeSpec\":{\"type\":\"value\",\"max\":2147483647,\"min\":0,\"scale\":0,\"step\":1,\"unit\":\"\"}},{\"abilityId\":117,\"accessMode\":\"rw\",\"code\":\"total_clean_area\",\"description\":\"\",\"name\":\"累计面积\",\"typeSpec\":{\"type\":\"value\",\"max\":2147483647,\"min\":0,\"scale\":0,\"step\":1,\"unit\":\"\"}},{\"abilityId\":118,\"accessMode\":\"rw\",\"code\":\"total_clean_count\",\"description\":\"\",\"name\":\"总次数\",\"typeSpec\":{\"type\":\"value\",\"max\":2147483647,\"min\":0,\"scale\":0,\"step\":1,\"unit\":\"\"}},{\"abilityId\":119,\"accessMode\":\"rw\",\"code\":\"side_brush_time\",\"description\":\"\",\"name\":\"边刷\",\"typeSpec\":{\"type\":\"value\",\"max\":270000,\"min\":0,\"scale\":0,\"step\":1,\"unit\":\"\"}},{\"abilityId\":120,\"accessMode\":\"rw\",\"code\":\"main_brush_time\",\"description\":\"\",\"name\":\"主刷\",\"typeSpec\":{\"type\":\"value\",\"max\":1080000,\"min\":0,\"scale\":0,\"step\":1,\"unit\":\"\"}},{\"abilityId\":121,\"accessMode\":\"rw\",\"code\":\"filter_time\",\"description\":\"\",\"name\":\"滤网\",\"typeSpec\":{\"type\":\"value\",\"max\":270000,\"min\":0,\"scale\":0,\"step\":1,\"unit\":\"\"}},{\"abilityId\":122,\"accessMode\":\"ro\",\"code\":\"robot_fault\",\"description\":\"\",\"extensions\":{\"scope\":\"fault\"},\"name\":\"机器故障\",\"typeSpec\":{\"type\":\"bitmap\",\"label\":[\"1\",\"2\",\"3\",\"4\",\"5\",\"6\",\"7\",\"8\",\"9\",\"10\",\"11\",\"12\",\"13\",\"14\",\"15\",\"16\",\"17\",\"18\",\"19\",\"20\",\"21\",\"22\",\"23\",\"24\",\"25\",\"26\",\"27\",\"28\",\"29\",\"30\"],\"maxlen\":30}},{\"abilityId\":123,\"accessMode\":\"rw\",\"code\":\"path_comm\",\"description\":\"\",\"name\":\"路径下发上报\",\"typeSpec\":{\"type\":\"raw\",\"maxlen\":128}},{\"abilityId\":124,\"accessMode\":\"rw\",\"code\":\"cmd_comm\",\"description\":\"\",\"name\":\"指令下发\",\"typeSpec\":{\"type\":\"raw\",\"maxlen\":128}},{\"abilityId\":125,\"accessMode\":\"rw\",\"code\":\"request_data\",\"description\":\"\",\"name\":\"请求数据\",\"typeSpec\":{\"type\":\"enum\",\"range\":[\"map\",\"path\",\"both\"]}},{\"abilityId\":126,\"accessMode\":\"rw\",\"code\":\"comm_flag\",\"description\":\"\",\"name\":\"简单通讯\",\"typeSpec\":{\"type\":\"enum\",\"range\":[\"1\"]}},{\"abilityId\":127,\"accessMode\":\"rw\",\"code\":\"comm_raw\",\"description\":\"指哪扫那 区域 禁区 其他\",\"name\":\"复杂通讯\",\"typeSpec\":{\"type\":\"raw\",\"maxlen\":128}},{\"abilityId\":128,\"accessMode\":\"ro\",\"code\":\"message_report\",\"description\":\"\",\"name\":\"消息上报\",\"typeSpec\":{\"type\":\"enum\",\"range\":[\"0\",\"1\",\"2\",\"3\",\"4\",\"5\",\"6\",\"7\",\"8\",\"9\",\"10\",\"11\",\"12\",\"13\",\"14\",\"15\",\"16\",\"17\",\"18\",\"19\",\"20\",\"21\",\"22\",\"23\",\"24\"]}},{\"abilityId\":129,\"accessMode\":\"wr\",\"code\":\"reset_map\",\"description\":\"\",\"name\":\"重置地图\",\"typeSpec\":{\"type\":\"bool\"}},{\"abilityId\":130,\"accessMode\":\"ro\",\"code\":\"sn\",\"description\":\"\",\"name\":\"设备序列号\",\"typeSpec\":{\"type\":\"raw\",\"maxlen\":128}},{\"abilityId\":131,\"accessMode\":\"ro\",\"code\":\"uuid\",\"description\":\"\",\"name\":\"UUID\",\"typeSpec\":{\"type\":\"raw\",\"maxlen\":128}},{\"abilityId\":132,\"accessMode\":\"rw\",\"code\":\"device_info\",\"description\":\"\",\"name\":\"获取设备信息\",\"typeSpec\":{\"type\":\"raw\",\"maxlen\":128}},{\"abilityId\":135,\"accessMode\":\"rw\",\"code\":\"yuliu\",\"description\":\"预留\",\"name\":\"预留\",\"typeSpec\":{\"type\":\"raw\",\"maxlen\":128}},{\"abilityId\":136,\"accessMode\":\"rw\",\"code\":\"yuliu1\",\"description\":\"预留\",\"name\":\"预留\",\"typeSpec\":{\"type\":\"raw\",\"maxlen\":128}},{\"abilityId\":137,\"accessMode\":\"rw\",\"code\":\"auto_boost\",\"description\":\"自动增压开关\",\"name\":\"地毯增压模式\",\"typeSpec\":{\"type\":\"bool\"}},{\"abilityId\":144,\"accessMode\":\"rw\",\"code\":\"room_mode_switch\",\"description\":\"\",\"name\":\"定制模式\",\"typeSpec\":{\"type\":\"bool\"}}]}]}"
`

**Comment by make-all** [comment, marked as answer]
Since you included the log, I can see that your device is missing the "Room mode switch" compared with the original. Making that optional in the config should let it work.

**Reply by mplab-casa** [reply]
Thanks a lot!

**Reply by borgotech** [reply]
Hi,
i also have a Rowenta x-plorer s75 animal and i want to integrate into HomeAssistant local tyya, i'm  new & newbie .... can you help me. I saw the previous discussion and I would like to know how to add "Room mode switch" in yaml file , can you explain me a little .. an example would be welcome.
Best Regards
